src/query/graphite/native/aggregation_functions.go

```diff
@@ -135,6 +135,39 @@ func divideSeries(ctx *common.Context, dividendSeriesList, divisorSeriesList sin
 	return r, nil
 }
 
+// aggregate takes a list of series and returns a new series containing the
+// value aggregated across the series at each datapoint using the specified function.
+func aggregate(ctx *common.Context, series singlePathSpec, fname string) (ts.SeriesList, error) {
+	switch fname {
+	case Empty, Sum, SumSeries, Total:
+		return sumSeries(ctx, multiplePathSpecs(series))
+	case Min, MinSeries:
+		return minSeries(ctx, multiplePathSpecs(series))
+	case Max, MaxSeries:
+		return maxSeries(ctx, multiplePathSpecs(series))
+	case Avg, Average, AverageSeries:
+		return averageSeries(ctx, multiplePathSpecs(series))
+	case Multiply, MultiplySeries:
+		return multiplySeries(ctx, multiplePathSpecs(series))
+	case Diff, DiffSeries:
+		return diffSeries(ctx, multiplePathSpecs(series))
+	case Count, CountSeries:
+		return countSeries(ctx, multiplePathSpecs(series))
+	case Range, RangeOf, RangeOfSeries:
+		return rangeOfSeries(ctx, series)
+	case Last, KeepLastValue:
+		return keepLastValue(ctx, series, -1) // using default value, same as graphite-web python implementation.
```

Comment:
Hm, I don't think this is what "last" does for "aggregate" func. It looks like it takes the highest value per series and gives back a single series:
https://github.com/graphite-project/graphite-web/blob/3ad279df5cb90b211953e39161df416e54a84948/webapp/graphite/functions/safe.py#L99-L101

Reply:
OK I just played around with Grafana, and I am fairly certain that the last function does this:

if your seriesList is 
```
[
  [A, B, C], 
  [D, E, F],
  [G, H, I]
]
```

once aggregated with `last`, our output will be:
```
[G, H, I]
```

because `last` just returns the last value at each point in time